src/common/color.helper.ts

```diff
@@ -1,16 +1,18 @@
 import { range } from 'd3-array';
 import { scaleBand, scaleLinear, scaleOrdinal, scaleQuantile } from 'd3-scale';
 
+import { ColorSet } from '../utils/color-set';
 import { colorSets } from '../utils/color-sets';
+import { ScaleType } from '../utils/scale-type.enum';
 
 export class ColorHelper {
   scale: any;
-  scaleType: any;
-  colorDomain: any[];
+  scaleType: ScaleType;
+  colorDomain: string[];
   domain: any;
   customColors: any;
 
-  constructor(scheme, type, domain, customColors?) {
+  constructor(scheme: ColorSet | string, type: ScaleType, domain, customColors?) {
```

Comment:
Since `scheme` will be assigned an unit from `colorSets` which is of type `ColorSet` and will be checked against the type `string`, then `scheme` can either take the type `ColorSet` or `string`.